### ISSUE TITLE
ssh/tailssh: optimize double encryption

### ISSUE THREAD

**Issue by bradfitz** [tailscale/tailscale#4473]
SSH encrypts.
WireGuard encrypts.

No need for both, really. (similar to the LetsEncrypt HTTPS + WireGuard we ran into earlier)

We can make our SSH server offer "insecure" old/faster ciphersuites to clients over Tailscale when we know the peer and know it's Tailscale.

But most likely the client won't want use/offer such thigns. Fortunately we have the optional `tailscale ssh` client side where we can enable them.

For potential performance improvement down the road if the double encryption hurts.

/cc @maisem @mayakacz @raggi

**Comment by raggi** [comment]
On modern x64, aes is probably going to be close to optimal - folks used to downgrade to rc4 for speed, but that was in the bad ol' days and the option is gone. On arm64, it's likely that the chaha20-poly1305 is going to be faster. One of the biggest things we can do to make things faster is ensure users aren't turning on compression, as ssh's zlib usage is painfully slow - about 30mbit/s on big x64 - much slower than the crypto.

It seems tempting to offer something like a `tssh` alias that's comfortably nearby, and mostly applicable-argument compatible with ssh. If tssh reliably handles the expected ssh API, some users might even be willing to swap out ssh for tssh in their $PATH, especially if it falls back to normal ssh when the target is outside of the tailnet.

**Comment by raggi** [comment]
Hmm, slightly more crazy idea. I wonder if we can make a fake session socket?

**Comment by bradfitz** [comment]
> It seems tempting to offer something like a tssh alias that's comfortably nearby, and mostly applicable-argument compatible with ssh. If

That's `tailscale ssh`. But that's optional. You can always use any normal SSH client.

**Comment by raggi** [comment]
> > It seems tempting to offer something like a tssh alias that's comfortably nearby, and mostly applicable-argument compatible with ssh. If
> 
> That's `tailscale ssh` (it already exists). But that's optional. You can always use any normal SSH client.

Right, one of the places where speed is likely to matter most is use cases like `rsync`, I'm not sure offhand if rsync is portably happy with a multi-word transport command (passed to `-e`), so the thought to provide a short single word alias just adds potential convenience to that kind of flow.
